From 984497e40e16a4b1594db0d5a64b338012a981de Mon Sep 17 00:00:00 2001
From: Vincent Hsu <vincent.hsu@linaro.org>
Date: Thu, 8 May 2014 11:32:03 +0800
Subject: [PATCH] ODP API Updates

1. odp_schedule(NULL) -> odp_schedule(NULL, ODP_SCHED_WAIT);
2. ODP_SCHED_SYNC_NONE -> ODP_SCHED_SYNC_ATOMIC

Signed-off-by: Vincent Hsu <vincent.hsu@linaro.org>
---
 pcap-linux.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pcap-linux.c b/pcap-linux.c
index 4372be6..feebd20 100644
--- a/pcap-linux.c
+++ b/pcap-linux.c
@@ -6207,7 +6207,7 @@ pcap_odp_init(pcap_t *handle)
 	 * resource
 	 */
 	qparam.sched.prio  = ODP_SCHED_PRIO_DEFAULT;
-	qparam.sched.sync  = ODP_SCHED_SYNC_NONE;
+	qparam.sched.sync  = ODP_SCHED_SYNC_ATOMIC;
 	qparam.sched.group = ODP_SCHED_GROUP_DEFAULT;
 	snprintf(inq_name, sizeof(inq_name), "%i-pktio_inq_def",
 		 (int)handle->pktio);
@@ -6364,7 +6364,7 @@ pcap_read_odp(pcap_t *handle, int max_packets, pcap_handler callback,
 
 	for (n = 1; (n <= max_packets) || (max_packets < 0); n++) {
 		/* Use schedule to get buf from any input queue */
-		buf = odp_schedule(NULL);
+		buf = odp_schedule(NULL, ODP_SCHED_WAIT);
 		/* fill out pcap_header */
 		gettimeofday(&ts, NULL);
 		pcap_header.ts = ts;
-- 
1.8.5

